src/App.js

```diff
@@ -1,19 +1,134 @@
 import React from 'react';
 import './App.css';
-
+import TodoList from './components/TodoList/TodoList';
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+state = {
+  newTodo: '',
+  todos: preparedTodos,
+  user: '',
+  userError: false,
+  titleError: false,
+}
+
+handleChange = (event) => {
+  this.setState({
+    newTodo: event.target.value,
+    titleError: false,
+  });
+}
+
+handleSubmit = (event) => {
+  event.preventDefault();
+
+  if (this.state.newTodo.trim().length === 0 && this.state.user.length === 0) {
+    this.setState({
+      titleError: true,
+      userError: true,
+    });
+
+    return;
+  }
+
+  if (this.state.user.length === 0) {
+    this.setState({ userError: true });
+
+    return;
+  }
+
+  if (this.state.newTodo.trim().length === 0) {
+    this.setState({ titleError: true });
+
+    return;
+  }
+
+  this.setState(state => ({ todos: [
+    {
+      completed: false,
+      title: state.newTodo,
+      user: users.find(person => person.name === state.user),
+      userId: users.find(person => person.name === state.user).id,
```

Comment:
1) why you need user AND userId, if you have user , you also have an id

Reply:
2. you can find user in advance and use it here, to prevent double invoke of find method